### PR TITLE
[rocm-openmp-extras] - Add support for flang-legacy in 6.1.2

### PR DESCRIPTION
**Pull request by estewart08**

### CODE REVIEW

**Comment by renjithravindrankannath** [comment]
@spackbot fix style

**Review by renjithravindrankannath — APPROVED** [review]
LGTM

File: var/spack/repos/builtin/packages/rocm-openmp-extras/package.py [hunk 1]
```diff
@@ -585,22 +624,43 @@ def install(self, spec, prefix):
         ]
         components["flang-runtime"] += flang_common_args
 
-        build_order = ["aomp-extras", "openmp", "openmp-debug", "pgmath", "flang", "flang-runtime"]
```
Comment (srekolam) [inline var/spack/repos/builtin/packages/rocm-openmp-extras/package.py:588]: i think for older releases like 5.7.1 etc the new build order will break. when building for 5.71 i saw below error 

  >> 14936    CMake Error: The source directory "/tmp/root/spack-stage/spack-stage-rocm-openmp-extras-5.7.1-nvfbpsjthzkzntke5ybrmbzbxfozbgzu/spack-src/rocm-openmp-extras/flang/flang-legacy/17.0-4/llvm-legacy/llvm" does not exist.
Can we add if self.spec.satisfies("@:6.0"): and include the older build_order . 
     14937    Specify --help for usage, or press the help button on the CMake GUI.

**Review by srekolam — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/rocm-openmp-extras/package.py [hunk 2]
```diff
@@ -471,13 +471,18 @@ def install(self, spec, prefix):
...
+        if os.path.islink((os.path.join(bin_dir, "flang-legacy"))):
+            os.unlink(os.path.join(bin_dir, "flang-legacy"))
         if os.path.islink((os.path.join(lib_dir, "libdevice"))):
             os.unlink(os.path.join(lib_dir, "libdevice"))
         if os.path.islink((os.path.join(llvm_prefix, "lib-debug"))):
             os.unlink(os.path.join(llvm_prefix, "lib-debug"))
 
         os.symlink(os.path.join(omp_bin_dir, "flang1"), os.path.join(bin_dir, "flang1"))
         os.symlink(os.path.join(omp_bin_dir, "flang2"), os.path.join(bin_dir, "flang2"))
+        os.symlink(
```
Comment (srekolam) [inline var/spack/repos/builtin/packages/rocm-openmp-extras/package.py]: on 5.7.1, i see a link for flang-leagcy getting created inside install prefix for llvm-amdgpu/bin and when i execute like below-  /home/spack/opt/spack/linux-rhel9-skylake_avx512/gcc-11.3.1/rocm-openmp-extras-5.7.1-owmri3xcla2f65bth3gfxw5g52m4j7ke/bin/flang-legacy
bash: /home/spack/opt/spack/linux-rhel9-skylake_avx512/gcc-11.3.1/rocm-openmp-extras-5.7.1-owmri3xcla2f65bth3gfxw5g52m4j7ke/bin/flang-legacy: No such file or directory

Comment (estewart08) [reply var/spack/repos/builtin/packages/rocm-openmp-extras/package.py]: There should be a flang-legacy symlink in llvm-amdgpu/bin that points to the binary in rocm-openmp-extras/bin. This should only be done for releases >= 6.1 as well as the addition of flang-legacy in the components. I will have to update the patch. Thanks!

**Review by srekolam — APPROVED** [review]

**Comment by srekolam** [comment]
Thanks Ethan!!

**Review by tldahlgren — DISMISSED** [review]

**Comment by tldahlgren** [comment]
@spackbot fix style

**Review by tldahlgren — APPROVED** [review]

**Comment by renjithravindrankannath** [comment]
Hi @tldahlgren , Please let us know if there anything to do before merging this PR.

**Pull request merged**